ntbs-service/Pages/Notifications/Delete.cshtml.cs

```diff
@@ -0,0 +1,65 @@
+using Microsoft.AspNetCore.Mvc;
+using ntbs_service.Models;
+using ntbs_service.Models.Enums;
+using ntbs_service.Models.Validations;
+using ntbs_service.Pages_Notifications;
+using ntbs_service.Services;
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+
+namespace ntbs_service.Pages.Notifications
+{
+    public class DeleteModel : NotificationModelBase
+    {
+        public ValidationService  ValidationService { get; set; }
+        [BindProperty]
+        [MaxLength(150)]
+        [RegularExpression(
+            ValidationRegexes.CharacterValidationWithNumbersForwardSlashAndNewLine, 
+            ErrorMessage = ValidationMessages.StringWithNumbersAndForwardSlashFormat)]
+        public string DeletionReason { get; set; }
+
+        public DeleteModel(INotificationService service, IAuthorizationService authorizationService) : base(service, authorizationService)
+        {
+            ValidationService = new ValidationService(this);
+        }
+
+        public async Task<IActionResult> OnGetAsync(int id)
+        {
+            Notification = await service.GetNotificationAsync(id);
+            if (Notification == null)
+            {
+                return NotFound();
+            }
+
+            NotificationId = Notification.NotificationId;
+            if (Notification.NotificationStatus != NotificationStatus.Draft)
+            {
+                return RedirectToPage("/Notifications/Overview", new { id = NotificationId });

```

Comment:
As far as I can read here, if you navigate to a deleted notification (or reload the deleted page when on confirmation), it will still work.
We probably want to confirm with Nancy what behaviour we want here, but I would guess at either a 404 for a deleted notification (small change to NotificationRepo would do here), or return the same partial?

Reply:
not sure i can quite see the worry here? is that the bit you guys wanted to discuss?